platform/qt/src/mbgl/gl_functions.cpp

```diff
@@ -10,278 +10,416 @@ namespace platform {
 /* OpenGL ES 2.0 */
 
 void (* const glActiveTexture)(GLenum) = [](auto... args) {
+    if (!QOpenGLContext::currentContext())
+        return;
```

Comment:
Maybe it's better to let it crash here instead? If these functions are not initialized MapLibre will not be able to render anything.

Reply:
Then we can not run unit tests. It seems that some GL calls are done in core after context is cleaned and this is a nice workaround. I'd wait for Metal work to finish to properly fix this.

Reply:
Maybe you could wrap the check in a function that returns a bool and logs a warning?

Your call, but it may save someone a frustrating debugging session.

Reply:
We can't change return type, but I could add logging, although I'm a bit hesitant. Even this `if` may slow down the code actually.

Let me see if I can hack the tests then...

Reply:
OK, I managed to add a functionality to `headless_backend.cpp` to restore the context when doing cleanup so we do not crash. This is a more proper solution.